packages/integration/src/bitbucket/BitbucketFetchService.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import fetch, { Response, RequestInfo, RequestInit } from 'node-fetch';
+import pThrottle from 'p-throttle';
+import { durationToMilliseconds } from '@backstage/types';
+import { FetchConfig } from '../helpers';
+
+/**
+ * A function that is a wrapper for the `fetch` function in node-fetch.
+ * This will either be a direct call to `fetch` or a throttled version of it.
+ * @public
+ */
+export type BitbucketFetchFunction = (
+  url: RequestInfo,
+  init?: RequestInit,
+) => Promise<Response>;
+
+/**
+ * A service that provides a `fetch` function that can be used to make HTTP requests.
+ * This service can be configured to throttle the number of requests that can be made.
+ * @internal
+ */
+export class BitbucketFetchService {
```

Comment:
If you make a `@backstage/integration-bitbucket-node` package, it could have a

```ts
function createThrottledFetch(options: {
  inner: typeof fetch,
  throttleByHost: {
    [host in string]: ThrottleConfig;
  }
}): typeof fetch
```

or something like that, and make it like a "middleware-like" thing around the inner fetch, [like we do](https://github.com/backstage/backstage/blob/526d5f1101a299cd0a9b0fe43c22355ef4a6bec3/packages/core-app-api/src/apis/implementations/FetchApi/IdentityAuthInjectorFetchMiddleware.ts#L55) when we [create the fetchApi](https://github.com/backstage/backstage/blob/526d5f1101a299cd0a9b0fe43c22355ef4a6bec3/packages/core-app-api/src/apis/implementations/FetchApi/createFetchApi.ts#L36) for the frontend.

No need for caching fetch implementations in a map. Its internal state will be a `Map<string, typeof pThrottle>` instead.